### ISSUE TITLE
pip install warnings on eth-utils version

### ISSUE THREAD

**Issue by WyseNynja** [makerdao/arbitrage-keeper#4]
When running `pip3 install -r requirements.txt` with pip 18, I get this warning:

```
eth-keys 0.2.0b3 has requirement eth-utils<2.0.0,>=1.0.0-beta.2, but you'll have eth-utils 0.7.1 which is incompatible.
eth-keyfile 0.5.1 has requirement eth-utils<2.0.0,>=1.0.0-beta.1, but you'll have eth-utils 0.7.1 which is incompatible.
eth-tester 0.1.0b32 has requirement eth-utils<2.0.0,>=1.0.1, but you'll have eth-utils 0.7.1 which is incompatible.
```

Is there any reason 0.7.1 is pinned in requirements.txt?

**Comment by WyseNynja** [comment]
I think using something like pip-tools (https://pypi.org/project/pip-tools/) would make managing versions easier. I'm happy to make a PR for that if you are okay with it.

I would add a `requirements.in` that looks like the .txt, but without any version pins and then I'd run pip-compile requirements.in. I'd also do the same for requirements-dev

**Comment by reverendus** [comment]
Yeah, that definitely looks like a nice tool! Regarding pinned dependencies, they are there as we are currently relying on an old version of `web3.py` and also relying on some features of its dependencies. 

We are currently working on upgrading to the most recent version of `web3.py`, that should really minimize the need of pinning anything.

**Comment by WyseNynja** [comment]
Cool. Is there an issue tracking upgrading web3.py?

**Comment by reverendus** [comment]
@WyseNynja it has been already done for `pymaker` on a separate branch, undergoing some testing at the moment.

**Comment by reverendus** [comment]
So `web3.py` has been upgraded in `pymaker`.
